--- media/filters/stream_parser_factory.cc	2016-03-11 15:48:23.094972354 +0100
+++ media/filters/stream_parser_factory.cc	2016-03-11 15:50:55.593687596 +0100
@@ -325,22 +325,16 @@
     std::vector<CodecInfo::HistogramTag>* video_codecs) {
   switch (codec_info->type) {
     case CodecInfo::AUDIO:
+      if (codec_info->tag == CodecInfo::HISTOGRAM_OPUS) {
+        return false;
+      }
       if (audio_codecs)
         audio_codecs->push_back(codec_info->tag);
       return true;
     case CodecInfo::VIDEO:
-#if defined(OS_ANDROID)
-      // VP9 is only supported on KitKat+ (API Level 19).
-      if (codec_info->tag == CodecInfo::HISTOGRAM_VP9 &&
-          base::android::BuildInfo::GetInstance()->sdk_int() < 19) {
-        return false;
-      }
-      // Opus is only supported on Lollipop+ (API Level 21).
-      if (codec_info->tag == CodecInfo::HISTOGRAM_OPUS &&
-          base::android::BuildInfo::GetInstance()->sdk_int() < 21) {
+      if (codec_info->tag == CodecInfo::HISTOGRAM_VP9) {
         return false;
       }
-#endif
       if (video_codecs)
         video_codecs->push_back(codec_info->tag);
       return true;
