--- content/gpu/gpu_child_thread.cc	2017-09-26 14:47:47.744506445 +0000
+++ content/gpu/gpu_child_thread.cc	2017-09-26 14:55:51.550279631 +0000
@@ -41,6 +41,8 @@
 #include "ui/gl/init/gl_factory.h"
 #include "url/gurl.h"
 
+#include "cef/libcef/renderer/media/gpu/cef_gpu_media_service.h"
+
 #if defined(USE_OZONE)
 #include "ui/ozone/public/gpu_platform_support.h"
 #include "ui/ozone/public/ozone_platform.h"
@@ -313,6 +315,7 @@
 
 void GpuChildThread::DidDestroyChannel(int client_id) {
   media_service_->RemoveChannel(client_id);
+  cef_gpu_media_service_->RemoveChannel(client_id);
   Send(new GpuHostMsg_DestroyChannel(client_id));
 }
 
@@ -398,6 +401,7 @@
                             sync_point_manager, gpu_memory_buffer_factory_));
 
   media_service_.reset(new media::MediaService(gpu_channel_manager_.get()));
+  cef_gpu_media_service_.reset(new CefGpuMediaService(gpu_channel_manager_.get()));
 
 #if defined(USE_OZONE)
   ui::OzonePlatform::GetInstance()
@@ -518,6 +522,7 @@
       params.client_id, params.client_tracing_id, params.preempts,
       params.allow_view_command_buffers, params.allow_real_time_streams);
   media_service_->AddChannel(params.client_id);
+  cef_gpu_media_service_->AddChannel(params.client_id);
   Send(new GpuHostMsg_ChannelEstablished(channel_handle));
 }
 
@@ -555,6 +560,7 @@
   if (gpu_channel_manager_) {
     gpu_channel_manager_->DestroyAllChannels();
     media_service_->DestroyAllChannels();
+    cef_gpu_media_service_->DestroyAllChannels();
   }
 }
 
--- content/gpu/gpu_child_thread.h	2017-09-26 14:56:11.594352581 +0000
+++ content/gpu/gpu_child_thread.h	2017-09-26 14:56:38.874451787 +0000
@@ -42,6 +42,8 @@
 class TargetServices;
 }
 
+class CefGpuMediaService;
+
 namespace content {
 class GpuProcessControlImpl;
 class GpuWatchdogThread;
@@ -151,6 +153,8 @@
 
   std::unique_ptr<media::MediaService> media_service_;
 
+  std::unique_ptr<CefGpuMediaService> cef_gpu_media_service_;
+
   // Information about the GPU, such as device and vendor ID.
   gpu::GPUInfo gpu_info_;
 
--- ui/ozone/public/native_pixmap.h	2017-10-05 15:13:48.147429518 +0000
+++ ui/ozone/public/native_pixmap.h	2017-09-22 10:39:28.586826271 +0000
@@ -26,6 +26,7 @@
   NativePixmap() {}
 
   virtual void* /* EGLClientBuffer */ GetEGLClientBuffer() const = 0;
+  virtual void* GetNativeSurface() const = 0;
   virtual int GetDmaBufFd() const = 0;
   virtual int GetDmaBufPitch() const = 0;
   virtual gfx::BufferFormat GetBufferFormat() const = 0;

